rust/sdk-examples/src/common_steps/postgres_version_tracker_step.rs

```diff
@@ -55,20 +54,34 @@ where
             seen_versions: AHashMap::new(),
         })
     }
+}
 
-    fn update_last_success_batch(&mut self, current_batch: TransactionContext<T>) {
-        let mut new_prev_batch = current_batch;
-        // While there are batches in seen_versions that are in order, update the new_prev_batch to the next batch.
-        while let Some(next_version) = self.seen_versions.remove(&(new_prev_batch.end_version + 1))
-        {
-            new_prev_batch = next_version;
-        }
-        self.next_version = new_prev_batch.end_version + 1;
-        self.last_success_batch = Some(new_prev_batch);
+#[async_trait]
+impl<T> VersionTrackerStep<T> for PostgresLatestVersionProcessedTracker<T>
```

Comment:
this is nice! 

i think in the very short time, we want to publish the entire SDK to crates.io and docs.rs. Let's document the Trait here and `rustdoc` can produce a page for it.

Reply:
thanks! Will do. I'm expecting to make some changes to this trait over the next couple of days so I'll wait until I'm done iterating to do that :D